go-fuzz/internal/writerset/writerset.go

```diff
@@ -0,0 +1,98 @@
+// Package writerset implements a mechanism to add and remove writers from a construct
+// similar to io.MultiWriter. Vendored from github.com/stephens2424/writerset
+package writerset
+
+import (
+	"io"
+	"net/http"
+	"sync"
+)
+
+// WriterSet wraps multiple writers like io.MultiWriter, but such that individual
+// writers are easy to add or remove as necessary.
+type WriterSet struct {
+	m  map[io.Writer]chan error
+	mu *sync.Mutex
```

Comment:
Because storing by value is the default. It is faster, consumes less memory, generates less garbage, less code (no need to initialize) and removes questions like "is this thing shared? does anybody else holds a pointer to it?". You use pointers only when there is a need to use a pointer. E.g. you want to allocate and initialize something lazily, or you want to share an object (two objects hold a pointer to the same mutex). Here is no need to use a pointer here.